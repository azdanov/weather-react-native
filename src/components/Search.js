import React from 'react'
import { StyleSheet, TextInput, View } from 'react-native'
import PropTypes from 'prop-types'

export default class Search extends React.Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
  }

  static defaultProps = {
    placeholder: '',
  }

  state = {
    focused: false,
    text: '',
  }

  handleChangeText = text => {
    this.setState({ text })
  }

  handleSubmitEditing = () => {
    const { onSubmit } = this.props
    const { text } = this.state

    if (!text) return

    this.setState({ text: '' }, () => onSubmit(text))
  }

  handleFocus = () => {
    this.setState({ focused: true })
  }

  handleBlur = () => {
    this.setState({ focused: false })
  }

  render() {
    const { onSubmit, placeholder } = this.props
    const { focused, text } = this.state

    return (
      <View
        style={{
          ...styles.container,
          backgroundColor: focused
            ? 'hsla(210, 22%, 49%, 0.8)'
            : 'hsla(209, 23%, 60%, 0.7)',
          borderColor: focused
            ? 'hsla(210, 22%, 49%, 0.9)'
            : 'hsla(209, 23%, 60%, 0.8)',
        }}
      >
        <TextInput
          autoCorrect={false}
          onBlur={this.handleBlur}
          onChangeText={this.handleChangeText}
          onFocus={this.handleFocus}
          onSubmit={onSubmit}
          onSubmitEditing={this.handleSubmitEditing}
          placeholder={placeholder}
          placeholderTextColor="white"
          style={styles.textInput}
          testID="search"
          underlineColorAndroid="transparent"
          value={text}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 5,
    borderWidth: 0.5,
    height: 40,
    marginHorizontal: 40,
    marginTop: 20,
    paddingHorizontal: 10,
  },
  textInput: {
    color: '#F0F4F8',
    flex: 1,
  },
})
