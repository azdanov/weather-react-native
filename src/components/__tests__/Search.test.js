import React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import Search from '../Search'

test('should render', () => {
  const { toJSON } = render(<Search onSubmit={jest.fn()} />)

  expect(toJSON()).toMatchSnapshot()
})

test('should show placeholder', () => {
  const placeholder = 'test'
  const { toJSON } = render(<Search placeholder={placeholder} onSubmit={jest.fn()} />)

  expect(toJSON()).toMatchSnapshot()
})

test('should call onSubmit', () => {
  const mockOnSubmit = jest.fn()
  const input = 'testing'

  const { getByTestId, toJSON } = render(<Search onSubmit={mockOnSubmit} />)

  fireEvent.changeText(getByTestId('search'), input)
  fireEvent(getByTestId('search'), 'submit')

  expect(mockOnSubmit).toHaveBeenCalledTimes(1)
  expect(toJSON()).toMatchSnapshot()
})
