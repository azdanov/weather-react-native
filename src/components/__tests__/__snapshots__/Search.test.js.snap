// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onSubmit 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsla(209, 23%, 60%, 0.7)",
      "borderColor": "hsla(209, 23%, 60%, 0.8)",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "height": 40,
      "marginHorizontal": 40,
      "marginTop": 20,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onSubmit={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onSubmitEditing={[Function]}
    placeholder=""
    placeholderTextColor="white"
    style={
      Object {
        "color": "#F0F4F8",
        "flex": 1,
      }
    }
    testID="search"
    underlineColorAndroid="transparent"
    value="testing"
  />
</View>
`;

exports[`should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsla(209, 23%, 60%, 0.7)",
      "borderColor": "hsla(209, 23%, 60%, 0.8)",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "height": 40,
      "marginHorizontal": 40,
      "marginTop": 20,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onSubmit={[MockFunction]}
    onSubmitEditing={[Function]}
    placeholder=""
    placeholderTextColor="white"
    style={
      Object {
        "color": "#F0F4F8",
        "flex": 1,
      }
    }
    testID="search"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`should show placeholder 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsla(209, 23%, 60%, 0.7)",
      "borderColor": "hsla(209, 23%, 60%, 0.8)",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "height": 40,
      "marginHorizontal": 40,
      "marginTop": 20,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onSubmit={[MockFunction]}
    onSubmitEditing={[Function]}
    placeholder="test"
    placeholderTextColor="white"
    style={
      Object {
        "color": "#F0F4F8",
        "flex": 1,
      }
    }
    testID="search"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
